### DISCUSSION TITLE
How to run frontend on local and other things on Docker

### DISCUSSION THREAD

**Discussion by yashpreetbathla** [cvat-ai/cvat#2898] [Q&A] [answered]
Hey All , i wanted to know if I want to run cvat-ui on frontend and other things on docker , which and what permisions should i give and more importantly in which files. Can anyone please give any idea about it.

**Comment by azhavoro** [comment, marked as answer]
Hi, I would suggest using docker-compose config to run CVAT locally, I think this is the easiest way. https://github.com/openvinotoolkit/cvat/blob/develop/cvat/apps/documentation/installation.md#ubuntu-1804-x86_64amd64
Could you please describe your case, maybe I misunderstand you.

**Reply by yashpreetbathla** [reply]
Hi , i have run my app on docker and it is running fine , but the problem that i have is
 (I want to make changes in frontend of CVAT_UI) so for that i want that changes are shown instantly but for now i have to build docker image in order to see changes in CVAT-UI that takes alot of time.. So is it possible i want to run CVAT-UI locally just by doing npm start in my machine and use backend running in docker..
I hope you got my problem.

**Reply by yashpreetbathla** [reply]
If i want to make changes in UI , is there any better way to see reflected changes quickly rather than making(building) docker image for cvat-ui , because it takes around 1-2 minutes to see a single change.

**Reply by azhavoro** [reply]
Thanks for the explanation, just setup dev environment as described here https://github.com/openvinotoolkit/cvat/blob/develop/CONTRIBUTING.md#development-environment

**Reply by yashpreetbathla** [reply]
yeah thanks for this i did that but after this i was able to run cvat-ui and backend but do i have to install Redis and other things as well because it was not functioning and was showing a error of not connecting to DB.
![image](https://user-images.githubusercontent.com/32976561/109788382-45238a00-7c35-11eb-959a-124af301ee59.png)

**Reply by yashpreetbathla** [reply]
Is it possible to run frontend in development env and other things at docker ?

**Reply by bsekachev** [reply]
@yashpreetbathla 

Possible. Need to do four things:

0. Perform installation of frontend-related part (according to ``CONTRIBUTING.md``)
1. In ``cvat/cvat-ui/package.json`` specify another value for variable ``REACT_APP_API_URL`` (in my case it is ``http://192.168.0.202:8080``, in your may be ``http://localhost:8080`` [default value] ).
2. Setup CORS properly in sever configuration files (``cvat/settings/development.py``) or you will get UI error something like this:
```
Access to XMLHttpRequest at 'http://192.168.0.202:8080/api/v1/users/self' from origin 'http://localhost:3000' has been blocked by CORS policy
```

3. Rebuild the image and restart the container

Finally you will get two UI servers (the first in container, the second - dev server on your host) and one REST server (in container).

**Reply by yashpreetbathla** [reply]
> @yashpreetbathla
> 
> Possible. Need to do four things:
> 
> 1. Perform installation of frontend-related part (according to `CONTRIBUTING.md`)
> 2. In `cvat/cvat-ui/package.json` specify another value for variable `REACT_APP_API_URL` (in my case it is `http://192.168.0.202:8080`, in your may be `http://localhost:8080` [default value] ).
> 3. Setup CORS properly in sever configuration files (`cvat/settings/development.py`) or you will get UI error something like this:
> 
> ```
> Access to XMLHttpRequest at 'http://192.168.0.202:8080/api/v1/users/self' from origin 'http://localhost:3000' has been blocked by CORS policy
> ```
> 
> 1. Rebuild the image and restart the container
> 
> Finally you will get two UI servers (the first in container, the second - dev server on your host) and one REST server (in container).

Hey thank you so much @bsekachev , i did these things but was stuck on CORs stuff so i tried changing at server side but didnt got success so for now i Disabled CORS in chrome.
